src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -88,15 +87,62 @@
 
 @postgres_dialect.segment(replace=True)
 class FunctionDefinitionGrammar(BaseSegment):
-    """This is the body of a `CREATE FUNCTION AS` statement."""
+    """This is the body of a `CREATE FUNCTION AS` statement.
+
+    Options supported as defined in https://www.postgresql.org/docs/9.1/sql-createfunction.html
+    """
 
     match_grammar = Sequence(
-        "AS",
-        OneOf(Ref("QuotedLiteralSegment"), Ref("DollarQuotedLiteralSegment")),
+        AnyNumberOf(
+            Sequence("TRANSFORM", "FOR", "TYPE", Ref("ParameterNameSegment")),
+            OneOf("WINDOW", "IMMUTABLE", "STABLE", "VOLATILE", "STRICT"),
+            Sequence(OneOf("NOT", optional=True), "LEAKPROOF"),
+            Sequence(
+                OneOf("CALLED", Sequence("RETURNS", "NULL")), "ON", "NULL", "INPUT"
+            ),
+            Sequence(
+                OneOf("EXTERNAL", optional=True),
```

Comment:
Use `Ref` instead of `OneOf` if you need to use the optional

```suggestion
                Ref("EXTERNAL", optional=True),
```

Reply:
Fixed

Reply:
Edit, not fixed. If I use Ref, both for this and NOT, it says that 
`
RuntimeError: Grammar refers to 'EXTERNAL' which was not found in the postgres dialect
`
Possibly a bug in Ref?

Reply:
Actually no, it seems `Ref` doesn't do what I thought it did when passed a literal value! `Ref.keyword` is what I should have said:

```suggestion
                Ref.keyword("EXTERNAL", optional=True),
```

Reply:
Fixed